packages/golden_toolkit/lib/src/multi_screen_golden.dart

```diff
@@ -84,3 +84,37 @@ Future<void> multiScreenGolden(
     );
   }
 }
+
+/// This [multiDeviceGolden] will create a [DeviceBuilder] for different [scenarios] for given [devices] list
+///
+/// Will output a single golden file for all devices
+///
+/// [name] is a file name output, must NOT include extension like .png
+///
+/// [devices] list of devices to run the tests - overriding `GoldenToolkit.configuration.defaultDevices`
+///
+/// [afterPump] is passed into DeviceBuilder scenario to be called after that device specific widget is rendered
+///
+Future<void> multiDeviceGolden(
```

Comment:
@coreysprague I agree we need to come up with a more concise API that pulls this functionality together. We have two interesting behaviors here. One is putting multiple "devices" in one golden, the other is giving each device in the golden unique or different states.

For the first behavior - this is an assumption but, I think if a team or individual is wanting large goldens like this that they probably want all of them like that. If that is true we might be able to make this a configuration. We could also add a way to override this behavior on a case by case basis. I am thinking of it as more of a capture mode.

For the second behavior - assuming we had a capture mode above... I think we could have an optional argument `deviceBuilder` that is mutually exclusive too `devices`. 

@moonytoes29 not asking you to make all these changes but I wanted to include you in the conversation regarding how we can improve the libraries API overall.

Reply:
@fatherOfLegends  - this `multiDeviceGolden` is definitely more of a helper for single scenario DeviceBuilder. Doesnt do much extra really. It was really to help me in my current project do a more simple find a replace rather than moving to the DeviceBuilder structure everywhere we already had a multiScreenGolden.

I think if we're looking for a more concise API in the future, we should expand more in the `GoldenBuilder`. I kept this separate as a `DeviceBuilder` to try and not to step on any toes. It could be another factory method under the `GoldenBuilder` namespace.

Another way i thought to tackle this whole PR, was to not create a large widget to encompass all the device sizes, but to just create the option to run any of the current golden methods in a method that just stitches the currently outputted pngs together into one png. But that seemed a lot more complicated.